x-pack/plugins/fleet/server/services/actions/client.test.ts

```diff
@@ -0,0 +1,287 @@
+  });
+
+  afterEach(() => {
+    appContextService.stop();
+  });
+
+  describe('create()', () => {
+    it('should create an action', async () => {
+      const action = generateFleetAction({ action_id: '1', input_type: 'foo' });
+      expect(await fleetActionsClient.create(action)).toEqual(action);
+      expect(mockedAuditLoggingService.writeCustomAuditLog).toHaveBeenCalledWith({
+        message: `User created Fleet action [id=1, user_id=${action.user_id}, input_type=foo]`,
+      });
+    });
+    it('should throw error when action does not match package name', async () => {
+      const action = generateFleetAction({ action_id: '1', input_type: 'bar' });
+      await expect(async () => await fleetActionsClient.create(action)).rejects.toBeInstanceOf(
+        FleetActionsClientError
+      );
+    });
+  });
+
+  describe('bulkCreate()', () => {
+    it('should bulk create actions', async () => {
+      const actions = [
+        {
+          action_id: uuidV4(),
+          input_type: 'foo',
+        },
+        {
+          action_id: uuidV4(),
+          input_type: 'foo',
+        },
+      ].map(generateFleetAction);
+      esClientMock.bulk.mockResolvedValue(generateFleetActionsBulkCreateESResponse(actions));
+
+      expect(await fleetActionsClient.bulkCreate(actions)).toEqual({
+        status: 'success',
+        items: actions.map((action) => ({
+          id: action.action_id,
+          status: 'success',
+        })),
+      });
+
+      expect(mockedAuditLoggingService.writeCustomAuditLog).lastCalledWith({
+        message: `User created Fleet action [id=${actions[1].action_id}, user_id=${actions[1].user_id}, input_type=foo]`,
+      });
+    });
+
+    it('should report errored documents', async () => {
+      const successActions = [
+        {
+          action_id: uuidV4(),
+          input_type: 'foo',
+        },
+        {
+          action_id: uuidV4(),
+          input_type: 'foo',
+        },
+      ].map(generateFleetAction);
+      const failedActions = [
+        {
+          action_id: uuidV4(),
+          input_type: 'foo',
+        },
+        {
+          action_id: uuidV4(),
+          input_type: 'foo',
+        },
+        {
+          action_id: uuidV4(),
+          input_type: 'foo',
+        },
+      ].map(generateFleetAction);
+
+      esClientMock.bulk.mockResolvedValue(
+        generateFleetActionsBulkCreateESResponse(successActions, failedActions, true)
+      );
+
+      expect(await fleetActionsClient.bulkCreate([...successActions, ...failedActions])).toEqual({
+        status: 'mixed',
+        items: successActions
+          .map((action) => ({
+            id: action.action_id,
+            status: 'success',
+          }))
+          .concat(
+            failedActions.map((action) => ({
+              id: action.action_id,
+              status: 'error',
+            }))
+          ),
+      });
+
+      // expect(mockedAuditLoggingService.writeCustomAuditLog).toHaveBeenCalledWith({
```

Comment:
nit: commented out code should be removed

Reply:
updated 19e31f859bf96727eb2d3d5b25b014044636c067